R/parse.R

```diff
@@ -1,24 +1,144 @@
 #' Parsing hms values
 #'
-#' These functions convert character vectors to objects of the [hms] class.
+#' This function converts character vectors to objects of the [hms] class.
 #' `NA` values are supported.
 #'
-#' `parse_hms()` accepts values of the form `"HH:MM:SS"`, with optional
-#' fractional seconds.
+#' `parse_hms()` accepts values of the form `"DdHH:MM:SS"`, with optional
+#' days, and fractional seconds. It also supports minus signs.
 #' @param x A character vector
 #' @export
 #' @examples
-#' parse_hms("12:34:56")
-#' parse_hms("12:34:56.789")
-parse_hms <- function(x) {
-  as.hms(as.difftime(as.character(x), format = "%H:%M:%OS", units = "secs"))
+#' time_str <- c("12:34:56",         # regular format
+#'               "12:34:56.789",     # decimal points
+#'               "20:12",            # HH:MM (no seconds)
+#'               "-12:34:56.001",    # negative values without days
+#'               "1d 12:34:56",      # number of days (integer)
+#'               "1d 12:34:56.001",  # decimal points and days
+#'               "-1d 12:34:56.001", # negative values
+#'               "-1d 12:34:56.001", # negative values and space
+#'               "0.5d",             # day only
+#'               NA                  # NA
+#' )
+#'
+#' parse_hms(time_str)
+parse_hms <- function(x){
+  out <- lapply(x, parse_one_hms)
+
+  issues <- sapply(out, function(time){
```

Comment:
Can you please copy over compat-purrr.R from rlang and use `map_lgl()` and friends?